shared/src/service/transport.ts

```diff
@@ -101,13 +102,25 @@ export class ClientTransportService implements TransportService {
   private authenticatedPromise: Promise<void>;
   private authenticated = false;
 
-  constructor(private clientId: string) {}
+  /**
+   * If useAccessToken is set, you will need to call setAccessToken on the
+   * ClientTransportService instance before making your first request and
+   * call it again whenever the token expires
+   */
+
+  constructor(private clientId: string, private useAccessToken?: boolean) {}
```

Comment:
I'm thinking that the signature here might need to be different. For one, I don't know that you need a flag for useAccessToken. You could simply have an accessToken private data member and a setter for it. The presence of a value would be the cue to avoid doing the authentication flow.

Also, clientId would be optional, since if you're providing the auth token, you don't need a clientId to do the OAuth flow with, and in our case, we won't have one.